crates/bevy_winit/src/lib.rs

```diff
@@ -42,15 +42,15 @@ impl Plugin for WinitPlugin {
     fn build(&self, app: &mut App) {
         app.init_resource::<WinitWindows>()
             .set_runner(winit_runner)
-            .add_system_to_stage(CoreStage::PostUpdate, change_window.exclusive_system());
+            .add_system_to_stage(CoreStage::PostUpdate, change_window);
     }
 }
 
-fn change_window(world: &mut World) {
-    let world = world.cell();
-    let winit_windows = world.get_resource::<WinitWindows>().unwrap();
-    let mut windows = world.get_resource_mut::<Windows>().unwrap();
-
+fn change_window(
+    winit_windows: Res<WinitWindows>,
+    mut windows: ResMut<Windows>,
+    mut window_dpi_changed_events: ResMut<Events<WindowScaleFactorChanged>>,
```

Comment:
I would use EventWriter here; if we ever optimize it we'll get the benefits automatically.

It also communicates intent a bit better.

Reply:
It is probably introducing ambiguity and some one-frame delays. From what I understand by default all the exclusive systems run at the start of the stage and this change makes it run in between. I would assume the same applies to `gilrs_event_system`.
I don't know how to make these systems run first without making them exclusive again or putting labels everywhere. Idk if putting labels everywhere is an acceptable solution, I don't think it is very maintainable.
For Reference:
- Ambiguity before this change:
  ```
  -- "bevy_winit::change_window" and "bevy_audio::audio_output::play_queued_audio_system<bevy_audio::audio_source::AudioSource>"
  ```
- Ambiguity after this change:
  ```
  -- "bevy_render::camera::camera::camera_system<bevy_render::camera::projection::OrthographicProjection>" and "bevy_winit::change_window"
  -- "bevy_render::camera::camera::camera_system<bevy_render::camera::projection::PerspectiveProjection>" and "bevy_winit::change_window"
  -- "bevy_ui::widget::text::text_system" and "bevy_winit::change_window"
  -- "bevy_ui::flex::flex_node_system" and "bevy_winit::change_window"
  -- "bevy_winit::change_window" and "bevy_text::text2d::text2d_system"
  ```

Reply:
Good point, Fixed.